target/linux/ath79/dts/ar7240_tplink_tl-wa830re-v1.dts

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "ar7240.dtsi"
+
+/ {
+	model = "TP-Link TL-WA830RE v1";
+	compatible = "tplink,tl-wa830re-v1", "qca,ar7240";
+
+	aliases {
+		led-boot = &led_system;
+		led-failsafe = &led_system;
+		led-running = &led_system;
+		led-upgrade = &led_system;
+		label-mac-device = &ath9k;
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 11 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+
+		wps {
+			label = "wps";
+			linux,code = <KEY_WPS_BUTTON>;
+			gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		pinctrl-names = "default";
+		pinctrl-0 = <&switch_led_pins>;
+
+		led_system: system {
+			label = "tp-link:green:system";
+			gpios = <&gpio 1 GPIO_ACTIVE_LOW>;
+		};
+
+		qss {
+			label = "tp-link:green:qss";
+			gpios = <&gpio 0 GPIO_ACTIVE_LOW>;
+		};
+
+		lan {
+			label = "tp-link:green:lan";
+			gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	ath9k-leds {
+		compatible = "gpio-leds";
+
+		wlan {
+			label = "tp-link:green:wlan";
+			gpios = <&ath9k 1 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+	};
+};
+
+&spi {
+	status = "okay";
+
+	num-cs = <1>;
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <104000000>;
```

Comment:
Really?

Reply:
Not 100% sure, the part from the datasheet ist

> - Normal READ (Serial): 40-MHz clock rate
> - FAST_READ (Serial): 104-MHz clock rate (maximum)
> - DUAL I/O FAST_READ: 80-MHz clock rate or 20 MB/s effective data rate
> - QUAD I/O FAST_READ: 80 MHz clock rate or 40 MB/s  effective data rate

and there are other boards with the same chip and this frequency, e.g. https://github.com/openwrt/openwrt/blob/master/target/linux/ath79/dts/ar9331_tplink_tl-mr3020-v1.dts#L123

Reply:
I also found some devices with ar7161, where they additionally set m25p,fast-read.
But to be honest, this is not my area of expertise, so I can't really tell what's correct here either.
I just was triggered by the relatively high value.

Is the device running stable with this frequency?

Reply:
I didn't test very long but it didn't crash. I addad `m25p,fast-read` and it seems to work, too.